### PR TITLE
Added --apply-namespace option

### PR DESCRIPTION
**Pull request by MNoeva**
#### Description: 
Added new parameters for namespaces;

https://github.com/cloudfoundry/multiapps/pull/265
https://github.com/cloudfoundry/multiapps-controller/pull/1492
https://github.com/SAP-samples/cf-mta-examples/pull/90

### CODE REVIEW

File: commands/deploy_command.go [hunk 1]
```diff
@@ -303,6 +312,15 @@ func (c *DeployCommand) executeInternal(positionalArgs []string, dsHost string,
 	// Build the process instance
 	processBuilder := NewDeploymentStrategy(flags, c.processTypeProvider).CreateProcessBuilder()
 	processBuilder.Namespace(namespace)
+	if(namespace == "") {
```
Comment (radito3) [inline commands/deploy_command.go:315]:
```suggestion
	if (len(namespace) == 0) {
```

File: commands/deploy_command.go [hunk 2]
```diff
@@ -609,12 +627,31 @@ type deployCommandFlagsValidator struct{}
 func (deployCommandFlagsValidator) ValidateParsedFlags(flags *flag.FlagSet) error {
 	var err error
 	flags.Visit(func(f *flag.Flag) {
-		if f.Name == strategyOpt {
+		switch f.Name {
```
Comment (radito3) [inline commands/deploy_command.go:630]: Does this need a default case?

File: commands/deploy_command.go [hunk 3]
```diff
@@ -609,12 +627,31 @@ type deployCommandFlagsValidator struct{}
...
+				err = errors.New("ApplyNamespaceServiceNames flag defined but no argument specified")
+			} else if f.Value.String() != "true" && f.Value.String() != "false" {
+				err = fmt.Errorf("invalid value for --%s, expected true or false", applyNamespaceServiceNamesOpt)
+			}
+		case applyNamespaceAppRoutesOpt:
+			if f.Value.String() == "" {
+				err = errors.New("ApplyNamespaceAppRoutes flag defined but no argument specified")
+			} else if f.Value.String() != "true" && f.Value.String() != "false" {
+				err = fmt.Errorf("invalid value for --%s, expected true or false", applyNamespaceAppRoutesOpt)
+			}
```
Comment (radito3) [inline commands/deploy_command.go:654]: These cases are very similar. Extract in method

**Review by radito3 — APPROVED** [review]

File: commands/deploy_command.go [hunk 4]
```diff
@@ -652,28 +673,13 @@ func (deployCommandFlagsValidator) ValidateParsedFlags(flags *flag.FlagSet) erro
...
-                err = e
-                return
-            }
-        case startTimeoutOpt:
-            if e := ValidateTimeoutOption(f.Name, flags, 259200); e != nil {
-                err = e
-                return
-            }
-        case stageTimeoutOpt:
+        case timeoutOpt, startTimeoutOpt, stageTimeoutOpt, uploadTimeoutOpt, taskExecutionTimeoutOpt:
```
Comment (ikasarov) [inline commands/deploy_command.go:676]: I think this was separate to allow different values for the validation 'maxAllowedVal'

**Review by ikasarov — DISMISSED** [review]

**Review by ikasarov — APPROVED** [review]

**Pull request merged**